lib/puppet/application/lookup.rb

```diff
@@ -297,11 +297,13 @@ def main
         type = options.include?(:type) ? Puppet::Pops::Types::TypeParser.singleton.parse(options[:type], scope) : nil
         result = Puppet::Pops::Lookup.lookup(keys, type, options[:default_value], use_default_value, merge_options, lookup_invocation)
         puts renderer.render(result) unless explain
-      rescue Puppet::DataBinding::LookupError
+      rescue Puppet::DataBinding::LookupError => e
+        lookup_invocation.report_text { e.message }
         exit(1) unless explain
       end
       puts format == :s ? lookup_invocation.explainer.to_s : renderer.render(lookup_invocation.explainer.to_hash) if explain
     end
+    exit(0)
```

Comment:
In the strictest sense this is an API break. The intent is that the lookup command line can be used in scripts and that it will error if not returning a value. If that feature should be carried over it can be supported with an extra flag controlling what it should do on conditions like 'not found'.